telluric/products.py

```diff
@@ -0,0 +1,840 @@
+        'nir': {'min': 750, 'mean': 825, 'max': 900},
+        # new micro def
+        'R450': {'min': 443, 'mean': 450, 'max': 457},
+        'R530': {'min': 521, 'mean': 530, 'max': 539},
+        'R550': {'min': 540, 'mean': 550, 'max': 560},
+        'R570': {'min': 560, 'mean': 570, 'max': 580},
+        'R582': {'min': 572, 'mean': 582, 'max': 592},
+        'R670': {'min': 657, 'mean': 670, 'max': 683},
+        'R690': {'min': 676, 'mean': 690, 'max': 703},  # should be confirmed with Martin
+        'R700': {'min': 686, 'mean': 700, 'max': 714},
+        'R750': {'min': 735, 'mean': 750, 'max': 765},
+        'R827': {'min': 809, 'mean': 827, 'max': 845},  # should be confirmed with Martin
+    }
+
+    @classmethod
+    def get_name(cls):
+        return cls.name
+
+    @classmethod
+    def get_order(cls):
+        return cls._order
+
+    @classmethod
+    def to_dict(cls):
+        attributes = "name min max required_bands output_bands description default_view type display_name unit".split()
+        d = {}
+        for attr in attributes:
+            d[attr] = getattr(cls, attr)
+        return d
+
+    def _apply(self, **bands):
+        raise NotImplementedError
+
+    @classmethod
+    def band_metadata(cls, band_name, metadata_source, meta, default=None):
+        band_meta = metadata_source.get(band_name, None)
+        if band_meta:
+            if meta:
+                return band_meta[meta]
+            else:
+                return band_meta
+        else:
+            return default
+
+    @classmethod
+    def max_wavelength(cls, sensor_bands_info, band_name, default=None):
+        return cls.band_metadata(band_name, sensor_bands_info, 'max', default)
+
+    @classmethod
+    def min_wavelength(cls, sensor_bands_info, band_name, default=None):
+        return cls.band_metadata(band_name, sensor_bands_info, 'min', default)
+
+    @classmethod
+    def min_product_wavelength(cls, band_name):
+        return cls.band_metadata(band_name, cls.wavelength_map, 'min')
+
+    @classmethod
+    def max_product_wavelength(cls, band_name):
+        return cls.band_metadata(band_name, cls.wavelength_map, 'max')
+
+    @classmethod
+    def match_bands(cls, sensor_bands_info, bands_list, dest_band):
+        min_val = cls.min_product_wavelength(dest_band)
+        max_val = cls.max_product_wavelength(dest_band)
+        return [a for a in bands_list if (min_val <= ceil(cls.min_wavelength(sensor_bands_info, a, default=0)) and
+                                          floor(cls.max_wavelength(sensor_bands_info, a, default=1000)) <= max_val)]
+
+    @classmethod
+    def fits_raster_bands(cls, available_bands, sensor_bands_info, bands_restriction=None, silent=True):
+        bands_restriction = bands_restriction or available_bands
+        matched_bands = cls.match_available_bands_to_required_bands(sensor_bands_info, bands_restriction)
+        if matched_bands:
+            do_we_have_elements_to_create_required_bands = all(
```

Comment:
I will need to explain what is an element, I am keep forgetting

Reply:
renamed `elements` to `input bands`